src/Compatibility/Microsoft.DotNet.ApiCompatibility/ValidationSuppressions/SuppressionEngine.cs

```diff
@@ -0,0 +1,208 @@
+        public void WriteSuppressionsToFile(string validationSuppressionFile)
+        {
+            XmlSerializer serializer = new XmlSerializer(typeof(Suppression[]));
+
+            using (Stream writer = GetWritableStream(validationSuppressionFile))
+            {
+                _readerWriterLock.EnterReadLock();
+                try
+                {
+                    serializer.Serialize(writer, _validationSuppressions.ToArray());
+                    AfterWrittingSuppressionsCallback(writer);
+                }
+                finally
+                {
+                    _readerWriterLock.ExitReadLock();
+                }
+            }
+        }
+
+        protected virtual void AfterWrittingSuppressionsCallback(Stream stream)
+        {
+            // Do nothing. Used for tests.
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="SuppressionEngine"/> based on the contents of a given suppression file.
+        /// </summary>
+        /// <param name="validationSuppressionFile">The path to the suppressions file to be used for initialization.</param>
+        /// <returns>An instance of <see cref="SuppressionEngine"/>.</returns>
+        public static SuppressionEngine CreateFromSuppressionFile(string validationSuppressionFile)
+            => new SuppressionEngine(validationSuppressionFile);
+
+        /// <summary>
+        /// Creates a new instance of <see cref="SuppressionEngine"/> which is empty.
+        /// </summary>
+        /// <returns>An instance of <see cref="SuppressionEngine"/>.</returns>
+        public static SuppressionEngine Create()
+            => new SuppressionEngine();
+
+        private HashSet<Suppression> ParseValidationSuppressionBaseline(string baselineFile)
+        {
+            HashSet<Suppression> result;
+            try
+            {
+                XmlSerializer serializer = new XmlSerializer(typeof(Suppression[]));
+                using (Stream reader = GetReadableStream(baselineFile))
+                {
+                    Suppression[]? deserializedSuppressions = serializer.Deserialize(reader) as Suppression[];
+                    if (deserializedSuppressions == null)
+                    {
+                        result = new HashSet<Suppression>();
+                    }
+                    else
+                    {
+                        result = new HashSet<Suppression>(deserializedSuppressions);
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                result = new HashSet<Suppression>();
+            }
+            return result;
+        }
+
+        protected virtual Stream GetReadableStream(string baselineFile) => new FileStream(baselineFile, FileMode.Open);
+
+        protected virtual Stream GetWritableStream(string validationSuppressionFile) => new FileStream(validationSuppressionFile, FileMode.OpenOrCreate);
```

Comment:
nit, use the same argument name for these two methods, it threw me off a little bit as I got confused thinking there was two different kind of files the suppression engine could accept.

Reply:
good catch, these are only for tests so not public surface area, but I agree that names should match.